Sources/ArcGISToolkit/Components/Augmented Reality/Joyslider.swift

```diff
@@ -0,0 +1,159 @@
+                                isChanging = false
+                                withAnimation(.bouncy.speed(1.5)) {
+                                    offset = 0
+                                }
+                                onEndedAction?()
+                            }
+                    )
+            }
+        }
+        .frame(height: Thumb.size + 4)
+        .task(id: isChanging) {
+            // This task block executes whenever isChanging property changes.
+            // If isChanging is `false`, then return.
+            guard isChanging else { return }
+            // Run a loop while the Task is not cancelled.
+            while !Task.isCancelled {
+                // Sleep for 50 milliseconds.
+                if #available(iOS 17, *) {
+                    try? await Task.sleep(for: .milliseconds(50))
+                } else {
+                    try? await Task.sleep(nanoseconds: 50_000_000)
+                }
+                // If task is cancelled after sleeping, return.
+                if Task.isCancelled { return }
+                // Otherwise change the value.
+                onChangedAction?(factor)
+            }
+        }
+    }
+    
+    /// Specifies an action to perform when the value changes.
+    func onChanged(perform action: @escaping (Double) -> Void) -> Joyslider {
+        var copy = self
+        copy.onChangedAction = action
+        return copy
+    }
+    
+    /// Specifies an action to perform when the value stops changing.
+    func onEnded(perform action: @escaping () -> Void) -> Joyslider {
+        var copy = self
+        copy.onEndedAction = action
+        return copy
+    }
+}
+
+/// Thumb view for the joyslider.
+private struct Thumb: View {
+    /// The size of the thumb.
+    static let size: Double = 26
+    
+    var body: some View {
+        Circle()
+            .stroke(Color.accentColor, lineWidth: 2)
+            .frame(width: Self.size, height: Self.size)
+            .background {
+                Circle()
+                    .frame(width: Self.size, height: Self.size)
+                    .foregroundStyle(.background)
+                    .shadow(color: .secondary.opacity(0.5), radius: 5, x: 0, y: 2)
+            }
+    }
+}
+
+/// Track view for the joyslider.
+private struct Track: View {
+    /// The x offset of the thumb in points.
+    let offset: Double
+    /// A factor between -1 and 1 that specifies the current percentage of the thumb's position.
+    let factor: Double
+    
+    var body: some View {
+        ZStack {
+            Capsule(style: .continuous)
+                .fill(Color.systemFill)
+            if offset != 0 {
+                Capsule(style: .continuous)
+                    .fill(fill)
+                    .frame(width: abs(offset))
+                    .offset(x: offset / 2)
+            }
+        }
+        .frame(height: 4)
+    }
+    
+    var fill: some ShapeStyle {
+        if offset >= 0 {
```

Comment:
This can also be done with a modifier such as `.scaleEffect(x: offset >= 0 ? 1: -1, y: 1)` that flips the capsule horizontally. I'm fine with the current implementation.